packages/fdc3-agent-proxy/src/messaging/AbstractMessaging.ts

```diff
@@ -102,15 +112,18 @@ export abstract class AbstractMessaging implements Messaging {
         this.implementationMetadata = null;
     }
 
-    getImplementationMetadata(): Promise<ImplementationMetadata> {
-        return Promise.resolve(this.implementationMetadata!!)
+    async getImplementationMetadata(): Promise<ImplementationMetadata> {
+        if(this.implementationMetadata === null){
```

Comment:
Reasonable to throw an error - where I hit this in another branch I generated impl metadata with "unknown" values. It shouldn't happen, but it pays to be defensive and to handle the error if it ever comes up.